src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getFruit() == null) {
+            throw new RuntimeException("There are no "
+                    + fruitTransaction.getFruit() + " int the storage");
+        }
+        int amountInStore = FruitStorage.getFruits().get(fruitTransaction.getFruit());
+        int amountToBuy = fruitTransaction.getQuantity();
+        if (amountInStore < amountToBuy) {
+            throw new RuntimeException("Not enough fruit in the storage for this operation");
+        }
+        FruitStorage.getFruits().put(fruitTransaction.getFruit(), amountInStore - amountToBuy);

```

Comment:
This line correctly updates the storage. However, if the `amountInStore` is equal to `amountToBuy`, the fruit should be removed from the map to reflect that it's no longer available in the storage.